pkg/scheduler/ugm/manager_test.go

```diff
@@ -702,6 +702,64 @@ func TestDecreaseTrackedResourceForGroupTracker(t *testing.T) {
+	assert.Equal(t, len(m.groupTrackers), 2, "Initial ugm state should have 2 groupTracker created.")
+
+	groupTracker1 := m.groupTrackers["group1"]
+	groupTracker2 := m.groupTrackers["group2"]
+
+	// Assert inital GroupTracker state
+	assert.Equal(t, len(groupTracker1.applications), 0, "Initial state shouldn't have any application in GroupTracker.")
+	assert.Equal(t, len(groupTracker2.applications), 0, "Initial state shouldn't have any application in GroupTracker.")
+	assert.Assert(t, groupTracker1.queueTracker.resourceUsage.IsEmpty(), "Initial state shouldn't have any resources tracked in GroupTracker.")
+	assert.Assert(t, groupTracker2.queueTracker.resourceUsage.IsEmpty(), "Initial state shouldn't have any resources tracked in GroupTracker.")
+
+	// Increase resources to different application, different queue
+	userGroup := security.UserGroup{User: user, Groups: groups}
+	usage, err := resources.NewResourceFromConf(map[string]string{"memory": "10", "vcore": "10"})
+	if err != nil {
+		t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage)
+	}
+	m.IncreaseTrackedResource("root.parent.leaf1", TestApp1, usage, userGroup)
+	m.IncreaseTrackedResource("root.parent.leaf2", TestApp2, usage, userGroup)
+
+	// Assert ugm post state
+	assert.Equal(t, len(m.userTrackers), 1, "Should have new userTrackers in ugm.")
+	assert.Equal(t, len(m.groupTrackers), 2, "groupTrackers in ugm should still be the same.")
+
+	// Assert UserTracker post state
+	userTracker := m.userTrackers[user]
+	assert.Assert(t, resources.Equals(userTracker.queueTracker.resourceUsage, resources.Multiply(usage, 2)), "The new increased resources should be tracked by UserTracker.")
+	assert.Equal(t, len(userTracker.appGroupTrackers), 2, "The two different GroupTracker should be referenced by userTracker.")
+	assert.Equal(t, userTracker.appGroupTrackers[TestApp1], groupTracker2, "UserTracker should hold application/GroupTracker mapping.")
+	assert.Equal(t, userTracker.appGroupTrackers[TestApp2], groupTracker1, "UserTracker should hold application/GroupTracker mapping.")
+
+	// Assert GroupTracker post state
+	assert.Assert(t, resources.Equals(groupTracker1.queueTracker.resourceUsage, usage), "Should have new resource tracked in GroupTracker.")
+	assert.Assert(t, resources.Equals(groupTracker2.queueTracker.resourceUsage, usage), "Should have new resource tracked in GroupTracker.")
+	assert.Equal(t, groupTracker1.applications[TestApp2], userGroup.User, "GroupTracker should hold applications/user mapping.")
+	assert.Equal(t, groupTracker2.applications[TestApp1], userGroup.User, "GroupTracker should hold applications/user mapping.")
+	assert.Assert(t, resources.Equals(groupTracker1.queueTracker.resourceUsage, usage), "The new increased resource should be tracked by GroupTracker.")
+	assert.Assert(t, resources.Equals(groupTracker2.queueTracker.resourceUsage, usage), "The new increased resource should be tracked by GroupTracker.")
```

Comment:
I can see a lot of asserts. Try to keep only mandatory checks or create a common method to do the assertion by passing necessary values as arguments.

Reply:
Most important thing to assert is, resource usage for either root.parent or root with different groups like group 1 and group2. In addition, there should not be group tracking for root.parent.leaf2, but group2 should be tracked for root.parent.leaf1

Reply:
Update to the simple test case as suggested below.  Remove the t.help functions I created in the last PR.

Reply:
> Most important thing to assert is, resource usage for either root.parent or root with different groups like group 1 and group2. In addition, there should not be group tracking for root.parent.leaf2, but group2 should be tracked for root.parent.leaf1

The current version: assert the 'root' queueTracker resourceUsage for different groups.
I remove the assertGroupTrackerQueuePathResource() function I created in the last PR to make things simpler.